src/deluge/modulation/arpeggiator.h

```diff
@@ -28,29 +28,116 @@ class ParamManagerForTimeline;
 
 typedef struct {
 	uint8_t length; // the number of steps to use, between 1 and 4
-	bool steps[4];  // the steps, whether they should play a note or a silence
+	bool steps[6];  // the steps, whether they should play a note or a silence
 } ArpRhythm;
 
-#define NUM_PRESET_ARP_RHYTHMS 8
+#define NUM_PRESET_ARP_RHYTHMS 51
 const ArpRhythm arpRhythmPatterns[NUM_PRESET_ARP_RHYTHMS] = {
-    {1, {1, 1, 1, 1}}, // <-
-    {3, {1, 1, 0, 1}}, // <-
-    {3, {1, 0, 1, 1}}, // <-
-    {4, {1, 1, 1, 0}}, // <-
-    {4, {1, 1, 0, 1}}, // <-
-    {4, {1, 0, 1, 1}}, // <-
-    {4, {1, 1, 0, 0}}, // <-
-    {4, {1, 0, 0, 1}}, // <-
+    {1, {1, 1, 1, 1, 1, 1}}, // <- 1 step
+    {2, {1, 0, 1, 1, 1, 1}}, // <- 2 steps
+    {3, {1, 1, 0, 1, 1, 1}}, // <- 3 steps
+    {3, {1, 0, 1, 1, 1, 1}}, // <-
+    {3, {1, 0, 0, 1, 1, 1}}, // <-
+    {4, {1, 1, 1, 0, 1, 1}}, // <- 4 steps
+    {4, {1, 1, 0, 1, 1, 1}}, // <-
+    {4, {1, 0, 1, 1, 1, 1}}, // <-
+    {4, {1, 1, 0, 0, 1, 1}}, // <-
+    {4, {1, 0, 0, 1, 1, 1}}, // <-
+    {4, {1, 0, 0, 0, 1, 1}}, // <-
+    {5, {1, 1, 1, 1, 0, 1}}, // <- 5 steps
+    {5, {1, 1, 1, 0, 1, 1}}, // <-
+    {5, {1, 1, 0, 1, 1, 1}}, // <-
+    {5, {1, 0, 1, 1, 1, 1}}, // <-
+    {5, {1, 1, 1, 0, 0, 1}}, // <-
+    {5, {1, 1, 0, 0, 1, 1}}, // <-
+    {5, {1, 0, 0, 1, 1, 1}}, // <-
+    {5, {1, 1, 0, 1, 0, 1}}, // <-
+    {5, {1, 0, 1, 0, 1, 1}}, // <-
+    {5, {1, 0, 1, 1, 0, 1}}, // <-
+    {5, {1, 1, 0, 0, 0, 1}}, // <-
+    {5, {1, 0, 0, 0, 1, 1}}, // <-
+    {5, {1, 0, 0, 1, 0, 1}}, // <-
+    {5, {1, 0, 1, 0, 0, 1}}, // <-
+    {5, {1, 0, 0, 0, 0, 1}}, // <-
+    {6, {1, 1, 1, 1, 1, 0}}, // <- 6 steps
+    {6, {1, 1, 1, 1, 0, 1}}, // <-
+    {6, {1, 1, 1, 0, 1, 1}}, // <-
+    {6, {1, 1, 0, 1, 1, 1}}, // <-
+    {6, {1, 0, 1, 1, 1, 1}}, // <-
+    {6, {1, 1, 1, 1, 0, 0}}, // <-
+    {6, {1, 1, 1, 0, 0, 1}}, // <-
+    {6, {1, 1, 0, 0, 1, 1}}, // <-
+    {6, {1, 0, 0, 1, 1, 1}}, // <-
+    {6, {1, 1, 1, 0, 1, 0}}, // <-
+    {6, {1, 1, 0, 1, 0, 1}}, // <-
+    {6, {1, 0, 1, 0, 1, 1}}, // <-
+    {6, {1, 0, 1, 1, 1, 0}}, // <-
+    {6, {1, 1, 1, 0, 0, 0}}, // <-
+    {6, {1, 1, 0, 0, 0, 1}}, // <-
+    {6, {1, 0, 0, 0, 1, 1}}, // <-
+    {6, {1, 1, 0, 0, 1, 0}}, // <-
+    {6, {1, 0, 0, 1, 0, 1}}, // <-
+    {6, {1, 1, 0, 1, 0, 0}}, // <-
+    {6, {1, 0, 1, 0, 0, 1}}, // <-
+    {6, {1, 1, 0, 0, 0, 0}}, // <-
+    {6, {1, 0, 0, 0, 0, 1}}, // <-
+    {6, {1, 0, 1, 0, 0, 0}}, // <-
+    {6, {1, 0, 0, 0, 1, 0}}, // <-
+    {6, {1, 0, 0, 0, 0, 0}}, // <-
 };
+
 const std::array<char const*, NUM_PRESET_ARP_RHYTHMS> arpRhythmPatternNames = {
-    "0",    // <-
-    "00-",  // <-
-    "0-0",  // <-
-    "000-", // <-
-    "00-0", // <-
-    "0-00", // <-
-    "00--", // <-
-    "0--0", // <-
+    "0",      // <- No rhythm: play all notes
```

Comment:
A lot of these are rotated versions of each other - should we split to a rythm+rotation instead?

Reply:
It would make managing it more difficult for the user. With just one knob (mappable to golden knobs) they can play with it directly

Reply:
I disagree - as a user I dislike that most of the rythms sound the same as I scroll through

Reply:
Some user requested it to be assignable to gold knobs, and although they sound similar it allows you to give exactly the vibe you want and also to play with it live with the knobs